jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -917,6 +917,12 @@ def _fit_model_to_cube(self, add_data):
 
         # Apply masks from selected spectral subset
         spec = self._apply_subset_masks(spec, self.spectral_subset)
+        # Also mask out NaNs for fitting. Simply adding filter_non_finite to the cube fit
+        # didn't work out of the box, so doing this for now.
+        if spec.mask is None:
```

Comment:
Is this new logic tested?

Reply:
The `None` case is, I should probably add a case to test the combination of existing mask and Nan-masking.

Reply:
I added a test that should cover the existing spec.mask as well (that mask would come from a spectral subset).